### PR TITLE
Fixed edge type property index lookup scan.

### PR DESCRIPTION
**Pull request by Darych**
Fixed edge type property index lookup scan.
Fixed disabling of edge type property index scan in case of more complex queries.
Why: queries which tries to construct a path based on edge type property index are returned as nulls.
Root cause: Edge type property index scan doesn't fill vertices in frame.

### CODE REVIEW

**Comment by Darych** [comment]
---
__*Leave above in PR description, copy the below into a comment*__
___


### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

**Comment by gitbuda** [comment]
@Darych thinking out loud here, is it true that the user will always get a plan that can be executed? In other words, it shouldn't be the case that if there is an index, we generate a plan that will throw an error preventing the user from executing anything. In other words, the suboptimal and correct plan is much better than the optimal plan throwing an exception.

File: src/query/plan/rewrite/edge_type_index_lookup.hpp [hunk 1]
```diff
@@ -599,19 +633,18 @@ class EdgeTypeIndexRewriter final : public HierarchicalLogicalOperatorVisitor {
...
+      switch (prop_filter_type) {
+        case PropertyFilter::Type::IS_NOT_NULL:
+        case PropertyFilter::Type::EQUAL: {
+          filter_exprs_for_removal_.insert(property_filter_->expression);
+          filters_.EraseFilter(*property_filter_);
+          return std::make_unique<ScanAllByEdgeTypeProperty>(input, output_symbol, edge_type_, *property_, view);
+        }
+        default: {
+          spdlog::warn("Filter is not supported by edge type property index.");
+        }
```
Comment (antoniofilipovic) [inline src/query/plan/rewrite/edge_type_index_lookup.hpp:646]: I don't think `spdlog::warn` is necessary here. The user can't do anything about it, so I would rather remove it.

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Small comment

**Comment by Darych** [comment]
> @Darych thinking out loud here, is it true that the user will always get a plan that can be executed? In other words, it shouldn't be the case that if there is an index, we generate a plan that will throw an error preventing the user from executing anything. In other words, the suboptimal and correct plan is much better than the optimal plan throwing an exception.

Yes, in this implementation we should fallback to not optimal ScanAll and return the result

File: src/query/plan/rewrite/edge_type_index_lookup.hpp (hunk 1 above)
Comment (Darych) [reply src/query/plan/rewrite/edge_type_index_lookup.hpp]: Thank you!

**Comment by kgolubic** [comment]
For release note I'll use:

- Fixed edge type property index lookup scan. The edge type property index scan
now correctly fills vertices in the frame, preventing queries that construct a
path based on edge type property index from returning nulls. The edge type
property index scan is disabled for more complex queries, defaulting to
`ScanAll` instead. [#2190](https://github.com/memgraph/memgraph/pull/2190)

**Pull request merged**